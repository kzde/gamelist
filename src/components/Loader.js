import React from 'react';
import styled, { keyframes } from 'styled-components';

const Overlay = styled.div`
  ${props => props.backgroundColor && `
  background-color: ${props.backgroundColor};
    position:absolute;
    top:0;
    left:0;
    right:0;
    bottom:0;
  `
  }
  
`;

const BounceCircle = styled.div`
  width: ${props => props.size || `40px`};
  height: ${props => props.size || `40px`};
  position: absolute;
  left: 50%;
  top: 50%;
`;

const bounce = keyframes`
  0% {transform: scale(0.0);}
  50% { transform: scale(1.0); }
  100% { transform: scale(0.0);}
`;
const Shared = styled.div`
    width: 100%;
    height: 100%;
    border-radius: 50%;
    background-color: ${props => props.color ? props.color : "black"};
    opacity: 0.6;
    position: absolute;
    top: 0;
    left: 0;
    animation: ${bounce} 3.0s infinite ease-in-out;
`;

const FirstCircle = styled(Shared)``;
const SecondCircle = styled(Shared)`
  animation-delay: -1.0s;
`

const Loader = (props) => (
  <Overlay backgroundColor={props.backgroundColor}>
    <BounceCircle size={props.size} >
      <FirstCircle color={props.color} />
      <SecondCircle color={props.color} />
    </BounceCircle>
  </Overlay>
)

export default Loader;