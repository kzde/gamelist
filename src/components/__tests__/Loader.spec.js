import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../Loader';

describe('Loader', () => {
  it('should renders correctly with default params', () => {
    const tree = renderer
      .create(<Loader />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should renders correctly with custom params', () => {
    const tree = renderer
      .create(<Loader color="white" size="60px" backgroundColor="black" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})