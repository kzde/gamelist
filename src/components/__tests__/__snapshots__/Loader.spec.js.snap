// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should renders correctly with custom params 1`] = `
<div
  className="Loader__Overlay-sc-1kdbz4n-0 cRLytn"
>
  <div
    className="Loader__BounceCircle-sc-1kdbz4n-1 lgORSW"
    size="60px"
  >
    <div
      className="Loader__Shared-sc-1kdbz4n-2 Loader__FirstCircle-sc-1kdbz4n-3 gyWvVa"
      color="white"
    />
    <div
      className="Loader__Shared-sc-1kdbz4n-2 Loader__SecondCircle-sc-1kdbz4n-4 gAbUBI"
      color="white"
    />
  </div>
</div>
`;

exports[`Loader should renders correctly with default params 1`] = `
<div
  className="Loader__Overlay-sc-1kdbz4n-0 jRrUCI"
>
  <div
    className="Loader__BounceCircle-sc-1kdbz4n-1 gMzLox"
    size="40px"
  >
    <div
      className="Loader__Shared-sc-1kdbz4n-2 Loader__FirstCircle-sc-1kdbz4n-3 clMjet"
      color="black"
    />
    <div
      className="Loader__Shared-sc-1kdbz4n-2 Loader__SecondCircle-sc-1kdbz4n-4 hpMKKM"
      color="black"
    />
  </div>
</div>
`;
